File: trees/postOrder/postOrder.pas
// Definimos un procedimiento llamado "postOrder" que toma un puntero a un nodo de árbol (de tipo "arbol") como argumento
procedure postOrder(nodo: arbol);
begin
    // Si el nodo no es nulo (es decir, existe), realizamos el recorrido en postorden
    if (nodo <> nil) then
    begin
        // Llamamos recursivamente a "postOrder" para el subárbol izquierdo
        postOrder(nodo^.hi);
        // Luego, llamamos recursivamente a "postOrder" para el subárbol derecho
        postOrder(nodo^.hd);
        // Imprimimos el valor del nodo actual
        writeln(nodo^.dato)
    end
end;

File: trees/postOrder/postOrder.ts
// Importamos el tipo Arbol desde su archivo correspondiente
import { Arbol } from "../type"

// Definimos una función llamada "postOrder" que toma un nodo raíz (de tipo Arbol o nulo) como argumento
export const postOrder = (root: Arbol | null): void => {
  // Si el nodo raíz no es nulo, realizamos el recorrido en postorden
  if (root !== null) {
    // Llamamos recursivamente a "postOrder" para el subárbol izquierdo
    postOrder(root.izquierdo)
    // Luego, llamamos recursivamente a "postOrder" para el subárbol derecho
    postOrder(root.derecho)
    // Imprimimos el valor del nodo actual
    console.log(root.valor)
  }
}
